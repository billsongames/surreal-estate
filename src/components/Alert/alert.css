.property-added-true {
  background-color: green;
  color: black;
}

.property-added-false {
  background-color: red;
  color: black;
}  