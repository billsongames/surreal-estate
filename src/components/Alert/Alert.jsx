import React from "react";
import PropTypes from "prop-types"

import "./alert.css"

function Alert ({ message, success }) {
  if(!message) return null
  
  return(
    <div className={`property-added-${success}`}>
      {message}
    </div>
  )
}    

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  success: PropTypes.bool
}

Alert.defaultProps = {
  success: false
}

export default Alert