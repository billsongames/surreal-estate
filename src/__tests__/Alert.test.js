import { render, screen } from '@testing-library/react';

import Alert from '../components/Alert/Alert';

describe("Alert tests", () => {
  test('Successful property alert', () => {

    render(<Alert message="Success!" success={true} />);
    expect(screen.getByText("Success!")).toBeInTheDocument()
});

  test('Failure property alert', () => {

    render(<Alert message="Error!" />);
    expect(screen.getByText("Error!")).toBeInTheDocument()
});

  test("no props does not render an alert mesage", () => {
    const {asFragment} = render(<Alert message="" />)
    expect(asFragment()).toMatchSnapshot()
  })

  test("Success props renders success alert mesage", () => {
    const {asFragment} = render(<Alert message="Property added successfully" />)
    expect(asFragment()).toMatchSnapshot()
  })

  test("Error props renders error alert mesage", () => {
    const {asFragment} = render(<Alert message="Error adding property" />)
    expect(asFragment()).toMatchSnapshot()
  })
})


