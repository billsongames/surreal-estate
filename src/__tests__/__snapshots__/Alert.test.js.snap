// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert tests Error props renders error alert mesage 1`] = `
<DocumentFragment>
  <div
    class="property-added-false"
  >
    Error adding property
  </div>
</DocumentFragment>
`;

exports[`Alert tests Success props renders success alert mesage 1`] = `
<DocumentFragment>
  <div
    class="property-added-false"
  >
    Property added successfully
  </div>
</DocumentFragment>
`;

exports[`Alert tests no props does not render an alert mesage 1`] = `<DocumentFragment />`;
